### PR TITLE
Server: add URL templating for event triggers and remote schemas

### LINKED ISSUE #2483: add action-style templating of webhook URL for event trigger and remote schemas
So far, when creating an event trigger and select a webhook url, we have the option to use environment variables but we cannot use string templates yet. Add support to be able to use string templates.

### PR DESCRIPTION
**Pull request by codingkarthik**
Add action-like URL templating for event triggers and remote schemas

fixes #2483 

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR. If no changelog is required, then add the `no-changelog-required` label.

### Affected components

- [x] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System
- [x] Tests
- [ ] Other (list it)

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [ ] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [x] No
- [ ] Yes
  - Does `run_sql` auto manages the new metadata through schema diffing?
    - [ ] Yes
    - [ ] Not required
  - Does `run_sql` auto manages the definitions of metadata on renaming?
    - [ ] Yes
    - [ ] Not required
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?
    - [ ] Yes
    - [ ] Not required

#### GraphQL
- [ ] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [x] No Breaking changes
- [ ] There are breaking changes:

  1. Metadata API

     Existing `query` types:
     - [ ] Modify `args` payload which is not backward compatible
     - [ ] Behavioural change of the API
     - [ ] Change in response `JSON` schema
     - [ ] Change in error code
     

  2. GraphQL API

     Schema Generation:
     
     
     - [ ] Change in any `NamedType`
     - [ ] Change in table field names
     

     Schema Resolve:-
     
     - [ ] Change in treatment of `null` value for any input fields 
     

  3. Logging

     - [ ] Log `JSON` schema has changed
     - [ ] Log `type` names have changed

### CODE REVIEW

File: scripts/dev.sh [hunk 1]
```diff
@@ -374,6 +374,7 @@ elif [ "$MODE" = "test" ]; then
   export EVENT_WEBHOOK_HEADER="MyEnvValue"
```
Comment (jberryman) [inline scripts/dev.sh:374]:
```suggestion
  # Various tests take some configuration from the environment; set these up here:
  export EVENT_WEBHOOK_HEADER="MyEnvValue"
```

File: server/src-lib/Hasura/RQL/Types/Common.hs [hunk 2]
```diff
@@ -305,6 +305,13 @@ instance FromJSON InputWebhook where
       Left e  -> fail $ "Parsing URL template failed: " ++ e
       Right v -> pure $ InputWebhook v
 
+instance Q.FromCol InputWebhook where
+  fromCol bs = do
+    urlTemplate <- parseURLTemplate <$> Q.fromCol bs
+    case urlTemplate of
+      Left e -> Left $ "Parsing URL template failed: " <> T.pack e
+      Right v -> Right $ InputWebhook v
```
Comment (jberryman) [inline server/src-lib/Hasura/RQL/Types/Common.hs:313]:
```suggestion
    bimap (\e-> "Parsing URL template failed: " <> T.pack e) InputWebhook urlTemplate
```
only if you want. You'll need to import `Data.Bifunctor` probably.

**Review by jberryman — APPROVED** [review]
LGTM! Just a couple suggestions you can take at your discretion

**Review by tirumaraiselvan — APPROVED** [review]
changelog

**Pull request merged**